c++/src/kj/compat/http.h

```diff
@@ -991,6 +991,19 @@ struct HttpClientSettings {
   // A reference to a TLS context that will be used when tlsStarter is invoked.
 };
 
+class WebSocketErrorHandler {
+public:
+  virtual kj::Exception handleWebSocketProtocolError(HttpHeaders::ProtocolError protocolError);
```

Comment:
Is HttpHeaders::ProtocolError the best f it for WebSocket errors? It looks like you're always passing the same values for all fields except `description`, so maybe just passing a string here would make sense?

Reply:
It might not be. I only need statusCode (it's usually 1002, but is 1009 at one call site) and description, so I could define a WebSocket-specific one with just those two fields if you prefer.

Reply:
Yeah I think it would make sense.

Reply:
Done in commit d817243.